### PR TITLE
Add text method to element

### PR DESCRIPTION
**Pull request by sujathakumar**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module github.com/iamneal/goleri
```
Comment (joente) [inline go.mod:1]: Can you exclude the go.mod file from the pull request?

File: result.go [hunk 2]
```diff
@@ -20,4 +21,21 @@ func (r *Result) GetExpecting() []Element {
 }
 
 // Tree returns the node tree.
-func (r *Result) Tree() *Node { return r.tree }
+func (r *Result) Tree() *Node {
+	r.assignExpressionParts()
```
[the remark this answers is not in the document]
Comment (iamneal) [reply result.go:25]: Ok we will take that out, thanks

**Review by joente — CHANGES_REQUESTED** [review]
Thanks for the pull request @iamneal!

Could you take a look at the test results? There are some lint issues with the pr. Otherwise it looks fine.

[Test CI](https://github.com/transceptor-technology/goleri/runs/4216855400?check_suite_focus=true) and [Test Lint](https://github.com/transceptor-technology/goleri/runs/4216855387?check_suite_focus=true)

**Comment by iamneal** [comment]
yeah I'll take a look

**Review by joente — APPROVED** [review]

**Comment by joente** [comment]
@iamneal ,

Maybe we could add a `GetNodeString` function to Result for retrieving a part of the string? 

Something like:

```go
// GetNodeString
func (r *Result) GetNodeString(n *Node) string {
	return r.expression[n.start:n.end]
}
```

This would require to parse the Node to the Result, but it keeps the Node struct light and does not unnecessary create a lot of new strings. (only when truly required).

**Pull request merged**